builder/format.go

```diff
@@ -64,3 +68,63 @@ func RevertTrimmedGroupsINI(b *Builder) error {
 	return helpers.CopyFile(filepath.Join(b.Config.Builder.ServerStateDir, "groups.ini"), filepath.Join(b.Config.Builder.ServerStateDir, "trimmed_groups.ini"))
 }
 
+func getLastBuildVersion(b *Builder) (string, error) {
+	// Likely the first build or some weirdness that requires us to get the container image and build it anyway
+	var lastVer []byte
+	var err error
+	filename := filepath.Join(b.Config.Builder.ServerStateDir, "image/LAST_VER")
+	if lastVer, err = ioutil.ReadFile(filename); err != nil {
+		return "", errors.Wrap(err, "Cannot find last built version, must get container image")
+	}
+	data := string(lastVer)
+	ver := strings.Split(data, "\n")
+
+	return ver[0], nil
+}
+
+// CheckBumpNeeded returns nil if it successfully deduces there is no format
+// bump boundary being crossed.
+func CheckBumpNeeded(b *Builder) (bool, error) {
+	version, err := getLastBuildVersion(b)
+	if err != nil {
+		return false, err
+	}
+	// Check what format our last built version is part of
+	versionURL := fmt.Sprintf("https://download.clearlinux.org/update/%s/format", version)
+	resp, err := http.Get(versionURL)
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to http.Get() %s", versionURL)
+	}
+	oldVer, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return false, errors.Wrapf(err, "Could not read format version from %s", versionURL)
+	}
+
+	// Check what format our to-be-built version is part of
+	versionURL = fmt.Sprintf("https://download.clearlinux.org/update/%s/format", b.MixVer)
+	resp, err = http.Get(versionURL)
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to http.Get() %s", versionURL)
+	}
+	newVer, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return false, errors.Wrapf(err, "Could not read format version from %s", versionURL)
+	}
+
+	// Check both formats are real numbers
```

Comment:
Is this actually a concern? Do we have released versions with bad format values?

Also, elsewhere we use the `parseUint32` in `builder` to parse versions as `uint32`. If you don't need these to be `uint64`, you could just reuse that function.

Reply:
If someone grabs a url like 20130 which doesn't actually exist since it was aborted it will correctly download you a 404 page, which is then compared to a (possibly) real number - in my testing at least... that's what happened.